eng/scripts/OpenAPI Hub/sync-repo-structure-cache.py

```diff
@@ -0,0 +1,118 @@
+from git import Repo
+from pymongo import MongoClient
+from typing import List
+import datetime
+import os
+import re
+
+
+def get_inner_blobs_structure(blob: Repo.tree) -> dict:
+    inner_structure: dict = {}
+    for inner_blob in blob:
+        if inner_blob.type == "tree":
+            inner_structure[inner_blob.name] = get_inner_blobs_structure(inner_blob)
+    return inner_structure
+
+
+def get_repo_tree(repo_url: str, clone_path: str, branch: str) -> [str, dict]:
+    print("Cloning to [{}]".format(clone_path))
+    if not os.path.isdir(clone_path):
+        print("Cloning repo into {} ...".format(clone_path))
+        repo: Repo = Repo.clone_from(repo_url, clone_path, branch=branch)
+    else:
+        print("Repo already cloned, checking out branch: {}".format(branch))
+        repo: Repo = Repo(clone_path)
+        repo.git.checkout(branch, force=True)
+
+    tree: Repo.tree = repo.tree()
+    print("Repo cloned successfully")
+
+    repo_structure: dict = {}
+    for blob in tree:
+        if blob.name == "specification" and blob.type == "tree":
+            repo_structure[blob.name] = get_inner_blobs_structure(blob)
+
+    if "specification" not in repo_structure:
+        raise Exception("No 'specification' folder found in the repo.")
+
+    return repo.head.commit.hexsha, repo_structure["specification"]
+
+
+def update_repo_structure_cache(
+    owner: str, repo_name: str, branch: str, github_token: str, repo_clone_path: str
+) -> dict:
+    new_document = {}
+    new_document["updated_on"] = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    new_document["key"] = "{}/{}/{}".format(owner, repo_name, branch)
+
+    repo_url: str = "https://{}@github.com/{}/{}".format(github_token, owner, repo_name)
+    clone_path: str = os.path.join(repo_clone_path, "{}{}".format(owner, repo_name))
+    new_document["commitSha"], new_document["repoStructure"] = get_repo_tree(
+        repo_url, clone_path, branch
+    )
+
+    return new_document
+
+
+def mongo_operations(
```

Comment:
`mongo_operations` is not a very descriptive function name 👍 

maybe `update_mongo_database` or something like that?